src/univariate/continuous/triangular.jl

```diff
@@ -93,20 +93,34 @@ entropy(d::TriangularDist{T}) where {T<:Real} = one(T)/2 + log((d.b - d.a) / 2)
 
 function pdf(d::TriangularDist, x::Real)
     a, b, c = params(d)
-    return if x < c
-        2 * max(x - a, 0) / ((b - a) * (c - a))
+    T = typeof(x / a)
```

Comment:
This line will error e.g. in the following case:
```julia
julia> pdf(TriangularDist{Int}(0, 2, 1), 0//1)
ERROR: DivideError: integer division error
Stacktrace:
 [1] div
   @ ./int.jl:288 [inlined]
 [2] divgcd(x::Int64, y::Int64)
   @ Base ./rational.jl:45
 [3] //
   @ ./rational.jl:65 [inlined]
 [4] /
   @ ./rational.jl:346 [inlined]
 [5] pdf(d::TriangularDist{Int64}, x::Rational{Int64})
   @ Main ./REPL[4]:3
 [6] top-level scope
   @ REPL[13]:1
```

Reply:
I've updated, which I believe solves the issue more generally as there won't be a div0 error anymore to interfere with getting the type.